pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/SourceConfigUtils.java

```diff
@@ -210,7 +212,21 @@ public static ExtractedSourceDetails validate(SourceConfig sourceConfig, Path ar
         if (!isEmpty(sourceConfig.getClassName())) {
             sourceClassName = sourceConfig.getClassName();
             try {
-                classLoader = Utils.extractClassLoader(archivePath, functionPkgUrl, uploadedInputStreamAsFile);
+                String ext = null;
+                if (!isEmpty(functionPkgUrl)) {
+                    File file = Utils.extractFileFromPkg(functionPkgUrl);
```

Comment:
this file is only a temporary file, we need it only for the extension.
can we remove the file as soon as we do not need it anymore ?
it would be also better to not extract the file but only to scan for the file name